lib/error_tracker/plugins/pruner.ex

```diff
@@ -0,0 +1,130 @@
+defmodule ErrorTracker.Plugins.Pruner do
+  @moduledoc """
+  Periodically delete resolved errors based on their age.
+
+  Pruning allows you to keep your database size under control by removing old errors that are not
+  needed anymore.
+
+  ## Using the pruner
+
+  To enable the pruner you must register the plugin in the ErrorTracker configuration. This will use
+  the default options, which is to prune errors resolved after 5 minutes.
+
+      config :error_tracker,
+        plugins: [ErrorTracker.Plugins.Pruner]
+
+  You can override the default options by passing them as an argument when registering the plugin.
+
+      config :error_tracker,
+        plugins: [{ErrorTracker.Plugins.Pruner, max_age: :timer.minutes(30)}]
+
+  ## Options
+
+  - `:limit`  - the maximum number of errors to prune on each execution. Occurrences are removed
+    along the errors. The default is 1000 to prevent timeouts and unnecesary database load.
+
+  - `:max_age` - the number of milliseconds after a resolved error may be pruned. The default is 5
```

Comment:
I would say 5 minutes is mostly the same as no time. A more sensible value could be 24h IMO.

At the end, resolved errors can arise again and you would lose all evidences if the pruner deletes it in 5-34 minutes

Reply:
Yeah, defaults is one thing that I wanted to discuss.

Ideally we would use sensible defaults that are useful for most people and don't require tinkering. I agree that 5 minutes is extremely low and 24h is much better.

Reply:
Updated this to 24h.